### DISCUSSION TITLE
Errors when calling ImGui::ShowDemoWindow();

### DISCUSSION THREAD

**Discussion by PolymorphicAgent** [ocornut/imgui#5348] [New Users Build/Link/Run issues ONLY!]
I am completely new to GUI's within C++, as all of my programs so far have been using STDIN and STDOUT. I do, however have a couple of years' experience coding in C++.
That being said, When I attempt to compile my program:
```
#include<bits/stdc++.h>
#include"../imgui.h"
int main(){
    ImGui::ShowDemoWindow();
}
```
With the command:
```
g++ main.cpp
```
I get the following error:
```
~/.../imgui/pgm$ g++ main.cpp
/tmp/ccXSwLkL.o: In function `main':
main.cpp:(.text+0xa): undefined reference to `ImGui::ShowDemoWindow(bool*)'
collect2: error: ld returned 1 exit status
```
Am I supposed to be adding certain compiler flags, or am I just making a stupidly obvious mistake?

**Comment by ocornut** [comment]
Undefined references mean you aren't linking the compiled object files containing those functions. You'll find ample references about this on the internet.

Once you manage to compile and link, you'll find calling `ImGui::ShowDemoWindow()' is absolutely not enough. Refer to the `examples/` application for a basic setup.

**Reply by PolymorphicAgent** [reply]
Figured it out, Now I'm excited!
